node/templates_test.go

```diff
@@ -0,0 +1,92 @@
+package node
+
+import (
+	"testing"
+
+	"github.com/NebulousLabs/Sia/siatest"
+)
+
+// TestNewNode is a basic smoke test for NewNode that uses all of the templates
+// to verify a working NewNode function.
+func TestNewNode(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	// Test AllModulesTemplate.
+	dir, err := siatest.TestDir("node", t.Name()+"-AllModulesTemplate")
+	if err != nil {
+		t.Fatal(err)
+	}
+	n, err := NewNode(AllModules(dir))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if n.Gateway == nil {
+		t.Error("gateway not set correctly")
+	}
+	if n.ConsensusSet == nil {
+		t.Error("consensus set not set correctly")
+	}
+	if n.Explorer == nil {
+		// TODO: Add the explorer to the node package.
+		t.Log("Need to add the explorer to the SiaNode framework.")
+	}
+	if n.TransactionPool == nil {
+		t.Error("transaction pool not set correctly")
+	}
+	if n.Wallet == nil {
+		t.Error("wallet not set correctly")
+	}
+	if n.Host == nil {
+		t.Error("host not set correctly")
+	}
+	if n.Renter == nil {
+		t.Error("renter not set correctly")
+	}
+	if n.Miner == nil {
+		t.Error("miner not set correctly")
+	}
+	err = n.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test WalletTemplate.
+	dir, err = siatest.TestDir("node", t.Name()+"-WalletTemplate")
+	if err != nil {
+		t.Fatal(err)
+	}
+	n, err = NewNode(Wallet(dir))
```

Comment:
Or maybe you only need a wallet. Then call `NewNode(Wallet(dir))`

Note that out of package, this would be `nodes.NewNode(nodes.Wallet(dir))`